activerecord/lib/active_record/relation/predicate_builder.rb

```diff
@@ -95,6 +95,11 @@ def create_binds_for_hash(attributes)
             next
           when value.is_a?(Relation)
             binds += value.bound_attributes
+          when value.is_a?(Array) && !table.type(column_name).respond_to?(:subtype)
+            if value.size == 1 && can_be_bound?(column_name, value.first)
+              result[column_name] = Arel::Nodes::BindParam.new
+              binds << build_bind_param(column_name, value.first)
```

Comment:
If RDBMSes supports an array type and the column is an array type, it is bound as an array value.

```ruby
  PgArray.find_by(tags: ["black"])
  # => SELECT  "pg_arrays".* FROM "pg_arrays" WHERE "pg_arrays"."tags" = $1 LIMIT $2  [["tags", "{black}"], ["LIMIT", 1]]
```